src/lib/components/composites/paper-components/PaperEntry.svelte

```diff
@@ -0,0 +1,91 @@
+    import { goto } from "$app/navigation";
+
+    interface PaperEntryProps {
+        paper: Paper;
+        projectId: number;
+        showReviewStatus?: boolean;
+        onClick?: () => void;
+    }
+
+    const navigateToPaperView = () => goto(`/project/${projectId}/paper/${paper.id}`);
+
+    const {
+        paper,
+        projectId,
+        showReviewStatus = false,
+        onClick = navigateToPaperView,
+    }: PaperEntryProps = $props();
+
+    // Mapping of review status to border color of paper entry
+    const reviewStatusColor: Record<ReviewStatus, string> = {
+        [ReviewProgress.Unreviewed]: "border-unreviewed-gray",
+        [ReviewDecision.Accepted]: "border-accept-green",
+        [ReviewDecision.Maybe]: "border-maybe-yellow",
+        [ReviewDecision.Declined]: "border-decline-red",
+    };
+
+    const showStatus = () => showReviewStatus && paper.reviewData !== undefined;
+
+    let timeoutId: ReturnType<typeof setTimeout> | null = null;
+    /**
+     * Handles the click event of the paper entry component by checking
+     * whether it was a single click (so no further click after 250ms) or a double click
+     * and call the corresponding functions:
+     *  - single click => onClick() (possible overridden, otherwise navigateToPaperView())
+     *  - double click => navigateToPaperView() (default)
+     *
+     * @param e - the click event
+     */
+    const handleClick = (e: MouseEvent) => {
+        if (e.detail === 1) {
+            if (timeoutId !== null) clearTimeout(timeoutId);
+            timeoutId = setTimeout(() => onClick(), 250);
+        } else if (e.detail === 2) {
+            if (timeoutId !== null) clearTimeout(timeoutId);
+            navigateToPaperView();
+        }
+    };
+</script>
+
+<!--
+@component
+Container displaying important paper information, optionally with review information.
+
+This component shows the
+  - paper id
+  - title
+  - authors (or 'unknown authors' if none are specified)
+as well as, if not in review mode, the review information about this paper.
+
+Furthermore this component is clickable and navigates to the corresponding paper view,
+if the event handler is not overridden.
+
+Usage:
+```svelte
+    <PaperEntry paper={paper} projectId={1} showReviewStatus={true} />
+```
+-->
+<button
+    type="button"
+    class="flex flex-row items-center justify-end pe-4 gap-3 w-full border {showStatus()
+        ? 'border-l-0'
+        : ''} border-container-border-grey rounded-md hover:bg-container-border-grey/35"
+    onclick={handleClick}
+>
+    <div
+        class="flex flex-auto {showStatus()
+            ? `border-l-4 ${reviewStatusColor[paper.reviewData?.reviewStatus ?? ReviewProgress.Unreviewed]}`
+            : ''} rounded-md px-3 py-1.5"
+    >
+        <PaperInfo {paper} />
+    </div>
+    {#if showStatus()}
+        {#each paper.reviewData!.individualDecisions as [user, individualDecision]}
```

Comment:
The `!` is not necessary here because TypeScript checks that `reviewData` is defined.
Alternatively, you can remove the `paper.reviewData !== undefined` check and use `paper.reviewData?.individualDecisions ?? []`.

Afaik, `!` is a bad practice because you fool the compiler to believe that something is defined.

Reply:
As showStatus() was a method, I got an error or at least a hint, that paper.reviewData could be null so I needed to add this check (or I was lost). But as the check is made via a variable now, it is completely fine to drop the ! and stick with best practice and not the bad practice.

Reply:
great!